eclass/cmake.eclass

```diff
@@ -28,12 +28,14 @@ if [[ -z ${_CMAKE_ECLASS} ]]; then
 _CMAKE_ECLASS=1
 
 # @ECLASS-VARIABLE: BUILD_DIR
+# @DEFAULT_UNSET
 # @DESCRIPTION:
 # Build directory where all cmake processed files should be generated.
 # For in-source build it's fixed to ${CMAKE_USE_DIR}.
 # For out-of-source build it can be overridden, by default it uses
-# ${WORKDIR}/${P}_build.
-: ${BUILD_DIR:=${WORKDIR}/${P}_build}
+# ${WORKDIR}/${P}_build (in EAPI-8: ${CMAKE_USE_DIR}_build).
```

Comment:
I think it's better to describe the situation for the newest EAPI first, then special cases for older EAPIs.